src/olympia/addons/views.py

```diff
@@ -735,6 +736,39 @@ def get_queryset(self):
         return qs.filter(addon=self.get_addon_object())
 
 
+class AddonReviewNotesViewSet(RetrieveModelMixin, ListModelMixin,
+                              GenericViewSet):
+    class AddonViewSetNoPublic(AddonViewSet):
+        permission_classes = [
+            AnyOf(AllowAddonAuthor, AllowReviewer, AllowReviewerUnlisted),
+        ]
+    permission_classes = [
+        AnyOf(AllowAddonAuthor, AllowReviewer, AllowReviewerUnlisted),
+    ]
+    serializer_class = ReviewNotesSerializer
+    queryset = Version.unfiltered.all()
+
+    def get_addon_object(self):
```

Comment:
Wow I missed that. I blame Mondays.

instead of keeping this `AddonViewSetNoPublic` you could pass `permission_classes` to it when instantiating it, with something like:

`self.addon_object = AddonViewSet(request=self.request, permission_classes=self.permission_classes, kwargs={'pk': self.kwargs['addon_pk']}).get_object()`. By doing that you could then move that code to a mixin.